CDP4Composition/Modularity/PluginUtilities.cs

```diff
@@ -26,27 +26,40 @@
 
 namespace CDP4Composition.Modularity
 {
+    using System;
     using System.Collections.Generic;
     using System.Diagnostics.CodeAnalysis;
     using System.IO;
+    using System.IO.Compression;
+    using System.Linq;
+    using System.Reflection;
 
     using CDP4Composition.Utilities;
 
     using Microsoft.Practices.ServiceLocation;
 
     using Newtonsoft.Json;
 
+    using NLog;
+
     /// <summary>
     /// Utility class that help with plugin management
     /// </summary>
     [ExcludeFromCodeCoverage]
     public static class PluginUtilities
     {
+        private static readonly Logger Logger = LogManager.GetCurrentClassLogger(typeof(PluginUtilities));
+
         /// <summary>
         /// The name of the plugin directory
         /// </summary>
         public const string PluginDirectoryName = "plugins";
 
+        /// <summary>
+        /// The directory of the downloaded plugin
+        /// </summary>
+        public const string DownloadDirectory = "RHEA/CDP4/DownloadCache/";
```

Comment:
does this not depend on where CDP4 is installed? how do we decide this magic string?

Reply:
I guess this is in AppData from code below, but indeed the first 2 folders should already be defined somewhere right?

also if you absolutely must provide nested folder paths make sure you use `Path.Combine` and not a hardcoded path seperator.

Reply:
yes these folder are in AppData and the structure of it was mentionned in the issue [#498](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues/498) also the folder `REAH/CDP4/` exists as it is used by the [AppSettingsServices](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/blob/fcf17eb552e88a1759117d48195203843e7090cc/CDP4Composition/Services/AppSettingService/AppSettingsService.cs#L44) Class. Shall refactor this so  `REAH/CDP4/` is defined correctly at the same place? 
```
/// <summary>
/// Gets the path of the AppData folder
/// </summary>
/// <returns>the Path the AppData folder</returns>
public static string GetAppDataFolder()
{
    var appDataPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),"RHEA", "CDP4");
    
    if (!Directory.Exists(appDataPath))
    {
        Directory.Create(appDataPath)
    }
    
   return appDataPath;
}
```